jwtAuth_test.go

```diff
@@ -78,12 +93,48 @@ func TestMiddleWare_StoreInformationInRequestcontext(t *testing.T) {
 	assert.Equal(t, "Alfred Bernard", storedClaims.DisplayName)
 }
 
+func TestMiddleWare_StoreInformationInRequestcontext_WithVerifyToken_WithExpiredToken_IgnoreExpiration(t *testing.T) {
+	jwtMiddleware := JwtAuthenticationMiddleware(fixtures.Fixtures.RawRsaPublicKey, &logrus.Logger{}, true, true)
+	modifiedRequest := &http.Request{}
+	fakeHandler := func(res http.ResponseWriter, req *http.Request) {
+		modifiedRequest = req
+	}
+
+	wrappedHandler := jwtMiddleware(fakeHandler)
+	headers := http.Header{"Authorization": {"Bearer " + fixtures.Fixtures.TokenExpired}}
+	initialRequest := &http.Request{Header: headers}
+	recorder := httptest.NewRecorder()
+	wrappedHandler(recorder, initialRequest)
+
+	storedClaims := modifiedRequest.Context().Value(tokenClaimsContextKey).(*TokenClaims)
+	assert.Equal(t, []Role{{Name: "cp:client:rider:"}}, storedClaims.Roles)
+	assert.Equal(t, "Carl De la Batte", storedClaims.DisplayName)
+}
+
+func TestMiddleWare_StoreInformationInRequestcontext_WithoutVerifyToken_WithExpiredToken(t *testing.T) {
+	jwtMiddleware := JwtAuthenticationMiddleware(fixtures.Fixtures.RawRsaPublicKey, &logrus.Logger{}, false, false)
```

Comment:
Actually, if the `verifyToken` is disabled it will mean that the publicKey is not sent as parameter and that will be the case for most of the `go` projects (if not all of them).

```suggestion
	jwtMiddleware := JwtAuthenticationMiddleware("", &logrus.Logger{}, false, false)
``